server/player/player.go

```diff
@@ -528,7 +528,10 @@ func (p *Player) Hurt(dmg float64, source damage.Source) (float64, bool) {
 		return 0, false
 	}
 
-	fireSource := source == damage.SourceFire{} || source == damage.SourceFireTick{} || source == damage.SourceLava{}
+	fireSource := source == damage.SourceFireTick{} || source == damage.SourceLava{}
+	if _, ok := source.(damage.SourceFire); ok {
```

Comment:
What's the point of this change?

Reply:
I need to type check instead of checking against a litteral value to since I needed it add a Campfire field within the SourceFire struct.

Reply:
I would probably type check for every other source then to be consistent.